examples_tests/02.ComputeShader/main.cpp

```diff
@@ -27,11 +38,584 @@ void main()
+			
+			if ((surfaceFormats.empty()) || (availablePresentModes == video::ISurface::EPM_UNKNOWN))
+				isGPUSuitable = false;
+
+			// Todo(achal): Probably a more sophisticated way to choose these
+			minSwapchainImageCount = core::min(surfaceCapabilities.minImageCount + 1u, MAX_SWAPCHAIN_IMAGE_COUNT);
+			if ((surfaceCapabilities.maxImageCount != 0u) && (minSwapchainImageCount > surfaceCapabilities.maxImageCount))
+				minSwapchainImageCount = surfaceCapabilities.maxImageCount;
+
+			surfaceFormat = surfaceFormats[0];
+			presentMode = static_cast<video::ISurface::E_PRESENT_MODE>(availablePresentModes & (1 << 0));
+			// preTransform = static_cast<nbl::video::ISurface::E_SURFACE_TRANSFORM_FLAGS>(surfaceCapabilities.currentTransform);
+			swapchainExtent = surfaceCapabilities.currentExtent;
+		}
+
+		if (isGPUSuitable) // find the first suitable GPU
+			break;
+	}
+	assert((computeFamilyIndex != ~0u) && (presentFamilyIndex != ~0u));
+
+	video::ILogicalDevice::SCreationParams deviceCreationParams;
+	if (computeFamilyIndex == presentFamilyIndex)
+	{
+		deviceCreationParams.queueParamsCount = 1u;
+		imageSharingMode = asset::ESM_EXCLUSIVE;
+	}
+	else
+	{
+		deviceCreationParams.queueParamsCount = 2u;
+		imageSharingMode = asset::ESM_CONCURRENT;
+	}
+
+	using QueueFamilyIndicesArrayType = core::smart_refctd_dynamic_array<uint32_t>;
+	QueueFamilyIndicesArrayType queueFamilyIndices;
+	queueFamilyIndices = core::make_refctd_dynamic_array<QueueFamilyIndicesArrayType>(deviceCreationParams.queueParamsCount);
+	{
+		const uint32_t tmp[] = { computeFamilyIndex, presentFamilyIndex };
+		for (uint32_t i = 0u; i < deviceCreationParams.queueParamsCount; ++i)
+			(*queueFamilyIndices)[i] = tmp[i];
+	}
+
+	// Could make this a static member of IGPUQueue, something like
+	// IGPUQueue::DEFAULT_QUEUE_PRIORITY, as a "don't-care" value for the user
+	const float defaultQueuePriority = 1.f;
+
+	std::vector<video::ILogicalDevice::SQueueCreationParams> queueCreationParams(deviceCreationParams.queueParamsCount);
+	for (uint32_t i = 0u; i < deviceCreationParams.queueParamsCount; ++i)
+	{
+		queueCreationParams[i].familyIndex = (*queueFamilyIndices)[i];
+		queueCreationParams[i].count = 1u;
+		queueCreationParams[i].flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+		queueCreationParams[i].priorities = &defaultQueuePriority;
+	}
+	deviceCreationParams.queueCreateInfos = queueCreationParams.data();
+
+	core::smart_refctd_ptr<video::ILogicalDevice> device = gpu->createLogicalDevice(deviceCreationParams);
+
+	video::IGPUQueue* computeQueue = device->getQueue(computeFamilyIndex, 0u);
+	video::IGPUQueue* presentQueue = device->getQueue(presentFamilyIndex, 0u);
+
+	// Todo(achal): We might want to check if: VK_FORMAT_FEATURE_STORAGE_IMAGE_BIT and
+	// VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT are supported for current surface format
+	// and current physical device. In fact, we might want to put this up as a criteria
+	// for making the physical device suitable
```

Comment:
report the format support (as a bitfield short vector) in one of the structures here
https://github.com/Devsh-Graphics-Programming/Nabla/blob/apis_as_plugins/include/nbl/video/IPhysicalDevice.h#L21

Reply:
Resolved